scripts/pylib/twister/expr_parser.py

```diff
@@ -264,6 +264,13 @@ def ast_expr(ast, env, edt):
 
         return ast_handle_dt_enabled_alias_with_parent_compat(edt, alias,
                                                               compat)
+    elif ast[0] == "dt_compat_enabled_with_label":
+        compat = ast[1][0]
+        label = ast[1][1]
+        for node in edt.nodes:
+            if node.status == "okay" and label in node.labels and node.matching_compat == compat:
+                return True
+        return False
```

Comment:
I'm late to the party, but this should use `edt.label2node` instead for speed. Something like this (untested):

```py
node = edt.label2node.get(label)
return node is not None and node.status == 'okay' and node.matching_compat == compat
```

@jfischer-no can you send a follow-up?

Reply:
@mbolivar-nordic thanks for the hint, see #33387